### DISCUSSION TITLE
string manipulation with snake, camel and dash to work same as their counterparts at lodash

### DISCUSSION THREAD

**Discussion by wouter-leistra** [rayepps/radash#96] [Ideas]
Hi,

Thanks for the great effort of making a modern typed lodash variant. We were eager to use it in our project, but we noticed some things that kept us reaching back to using lodash. One of the most prominent reasons has been the snake, camal (should be camel as it is referring to camelCase) and dash (we are used to using kebab, but we can live with the new name too).

In lodash the counterparts (snakeCase, camelCase and kebabCase) allow converting any space-separated, snake-cased, kebab-cased and camel-cased string to convert to the case you choose to use the function for. 

Example

```js
const str1="Hello world";
const str2="helloWorld";
const str3="hello-world";
const str4="hello_world";

// What we expected
snake(str1) => "hello_world";
snake(str2) => "hello_world";
snake(str3) => "hello_world";
snake(str4) => "hello_world";

// What radash does
snake(str1) => "hello_world";
snake(str2) => "helloWorld";
snake(str3) => "hello-world";
snake(str4) => "hello_world";
```
Is this a conscious decision by the maintainers? Or are you open for PRs to support feature parity?

**Comment by sodiray** [comment]
I don't care too much about Lodash feature parody but that result for `str2` and `str3` in Radash is just plain wrong 😞  so ya I'd love to get a fix in for it.

**Comment by MarlonPassos-git** [comment]
hellow @rayepps @wouter-leistra  thank you so much for pointing out this problem, i just made a pr for them fixing this.

https://github.com/rayepps/radash/pull/114

Now just wait for some Maintainer to approve

**Reply by sodiray** [reply]
I'm looking now 👀

**Comment by sodiray** [comment]
This has been fixed and published in version `v8.0.3`
